add docs for `ini_parse_quantity` function

realFlowControl opened the pull request:
Hello there :vulcan_salute:

this will add the documentation for the `ini_parse_quantity` function as found in #1803 

Kind regards
Florian

Girgias commented:
Just landed [0f8b9eb49bd644f07175e9191d2cc146737446ad](https://github.com/php/php-src/commit/0f8b9eb49bd644f07175e9191d2cc146737446ad) to add support for ``0b`` and ``0o`` prefixes, wondering if it makes sense to add examples with prefixes in the docs? (hex has worked since forever)

devnexen commented on reference/info/functions/ini-parse-quantity.xml:
```diff
@@ -0,0 +1,111 @@
+      <para>
+       The ini shorthand to parse, must be a number followed by an optional multiplier.
+
+       Multiplier can be any of <literal>k</literal>/<literal>K</literal> (1024),
```
I would suggest
`The following multipliers are supported :
  <literal>k</literal>/<literal>K</literal> (1024),
...
`

devnexen commented on reference/info/functions/ini-parse-quantity.xml:
```diff
@@ -0,0 +1,111 @@
+  &reftitle.returnvalues;
+  <para>
+   Returns the interpreted size in bytes as &integer; on success, or
+   <literal>0</literal> in case of unparsable shorthand and raises a
```
`raises an <constant>E_WARNING</constant> level error` or something approaching

realFlowControl replied:
:heavy_check_mark:

cmb69 commented on reference/info/functions/ini-parse-quantity.xml:
```diff
@@ -0,0 +1,111 @@
+      <para>
+       The ini shorthand to parse, must be a number followed by an optional multiplier.
+
+       Multiplier can be any of <literal>k</literal>/<literal>K</literal> (1024),
+       <literal>m</literal>/<literal>M</literal> (1048576),
+       <literal>g</literal>/<literal>G</literal> (1073741824).
```
I suggest to mark-up the numbers:
```suggestion
       Multiplier can be any of <literal>k</literal>/<literal>K</literal> (<literal>1024</literal>),
       <literal>m</literal>/<literal>M</literal> (<literal>1048576</literal>),
       <literal>g</literal>/<literal>G</literal> (<literal>1073741824</literal>).
```

cmb69 commented:
> wondering if it makes sense to add examples with prefixes in the docs?

I think it would make more sense, to add some `ini_get()` examples.

cmb69 commented on reference/info/functions/ini-parse-quantity.xml:
```diff
@@ -0,0 +1,111 @@
+   </example>
+  </para>
+ </refsect1>
+
```
```suggestion
 <refsect1 role="seealso">
  &reftitle.seealso;
  <simplelist>
   <member><function>ini_get</function></member>
  </simplelist>
 </refsect1>
```

devnexen commented:
Looks good but cc @cmb69

realFlowControl commented:
Hey there,
sorry for taking so long to come back to this PR. I think I incorporated all your hints :smile: 
Hope everything is fine, if there is something else, just ping.
@Girgias, @cmb69: I added the prefixes to the docs, I think it should be there for completeness.

Girgias commented on reference/info/functions/ini-parse-quantity.xml:
```diff
@@ -0,0 +1,122 @@
+ <refsect1 role="returnvalues">
+  &reftitle.returnvalues;
+  <para>
+   Returns the interpreted size in bytes as &integer; on success, or
```
```suggestion
   Returns the interpreted size in bytes as an &integer; on success, or
```

Girgias commented on reference/info/functions/ini-parse-quantity.xml:
```diff
@@ -0,0 +1,122 @@
+  </para>
+ </refsect1>
+
+ <refsect1 role="returnvalues">
+  &reftitle.returnvalues;
+  <para>
+   Returns the interpreted size in bytes as &integer; on success, or
+   raises a <constant>E_WARNING</constant> and returns a <literal>0</literal>
+   in case of unparsable shorthand.
+  </para>
+  <para>
+   In case of invalid multiplier the function will return the number and raise
+   a <constant>E_WARNING</constant>.
+  </para>
+ </refsect1>
```
```suggestion
 <refsect1 role="returnvalues">
  &reftitle.returnvalues;
  <para>
   Returns the interpreted size in bytes as an &integer;.
  </para>
 </refsect1>

 <refsect1 role="errors">
  &reftitle.errors;
  <para>
   If the value cannot be parsed, or an invalid multiplier is used, an <constant>E_WARNING</constant> is raised.
  </para>
 </refsect1>
```

Girgias commented on reference/info/functions/ini-parse-quantity.xml:
```diff
@@ -0,0 +1,122 @@
+   <example>
+    <title>A few <function>ini_parse_quantity</function> examples</title>
+    <programlisting role="php">
+<![CDATA[
+<?php
+
+var_dump(ini_parse_quantity('1024'));
+var_dump(ini_parse_quantity('1024M'));
+var_dump(ini_parse_quantity('512K'));
+var_dump(ini_parse_quantity('0xFFk'));
+var_dump(ini_parse_quantity('Foobar'));
+var_dump(ini_parse_quantity('10F'));
+
```
Maybe add some examples for the prefixes?

realFlowControl replied:
Thanks for the hint, I added two additional examples, one with binary and one octal.

Girgias reviewed (changes requested):
Not totally sure for the wording of the error section, but warnings should ideally not be mentioned in the return value section (yes the manual is not super consistent on this)

cmb69 commented on reference/info/functions/ini-parse-quantity.xml:
```diff
@@ -0,0 +1,127 @@
+   <methodparam><type>string</type><parameter>shorthand</parameter></methodparam>
+  </methodsynopsis>
+  <para>
+   Returns the interpreted size in bytes on success from an ini shorthand.
```
I think it makes sense to link to the explanation of ini shorthands:
```suggestion
   Returns the interpreted size in bytes on success from an <link linkend="faq.using.shorthandbytes">ini shorthand</link>.
```

cmb69 reviewed (commented):
Besides the missing link, this looks good to me.

realFlowControl commented on reference/info/functions/ini-parse-quantity.xml:
```diff
@@ -0,0 +1,127 @@
+       <literal>g</literal>/<literal>G</literal> (<literal>1073741824</literal>).
+
+       The number can be a decimal, hex (prefixed with <literal>0x</literal> or <literal>0X</literal>),
+       octal (prefixed with <literal>0o</literal> or <literal>0O</literal>) or binary (prefixed with
```
You are right, just added this in docs and example

cmb69 reviewed (approved):
Thank you!
